test/unit/functions.spec.ts

```diff
@@ -204,4 +204,20 @@ export class FunctionTests {
         Expect(() => transpiler.transpilePropertyCall(mockObject as ts.CallExpression))
             .toThrowError(Error, "Tried to transpile a non-property call as property call.");
     }
+
+    @Test("Function dead code after return")
+    public functionDeadCodeAfterReturn(): void {
+        const result = util.transpileAndExecute(
+            `function abc() { return 3; const a = 5; } return abc();`);
+
+        Expect(result).toBe(3);
+    }
+
+    @Test("Method dead code after return")
+    public methodDeadCodeAfterReturn(): void {
+        const result = util.transpileAndExecute(
+            `class def { public static abc() { return 3; const a = 5; } } return def.abc();`);
+
+        Expect(result).toBe(3);
+    }
```

Comment:
namespaces also use transpire block a test would be nice.

Reply:
You can not return from a namespace since it is not a function body.